mclagsyncd/mclaglink.cpp

```diff
@@ -401,276 +519,1390 @@ void MclagLink::setFdbEntry(char *msg, int msg_len)
+
+    cfg_msg_hdr = reinterpret_cast<mclag_msg_hdr_t *>(static_cast<void *>(infor_start));
+    cfg_msg_hdr->version = 1;
+    cfg_msg_hdr->msg_len = (unsigned short)infor_len;
+    cfg_msg_hdr->msg_type = MCLAG_SYNCD_MSG_TYPE_CFG_MCLAG_IFACE;
+
+    SWSS_LOG_DEBUG("mclagsycnd send msg to iccpd, msg_len =%d, msg_type =%d count : %d ver:%d ", cfg_msg_hdr->msg_len, cfg_msg_hdr->msg_type, cfg_msg_hdr->version, count);
+    write = ::write(getConnSocket(), infor_start, cfg_msg_hdr->msg_len);
+
+
+    if (write <= 0)
+    {
+        SWSS_LOG_ERROR("mclagsycnd to ICCPD, mclag iface cfg send; write to m_connection_socket failed");
+    }
+    return;
+}
+
+void MclagLink::mclagsyncdSendMclagUniqueIpCfg(std::deque<KeyOpFieldsValuesTuple> &entries)
+{
+    struct mclag_unique_ip_cfg_info cfg_info;
+    mclag_msg_hdr_t *cfg_msg_hdr = NULL;
+    size_t infor_len = sizeof(mclag_msg_hdr_t);
+    int count = 0;
+
+    ssize_t write = 0;
+    char *infor_start = getSendMsgBuffer();
+
+    /* Nothing popped */
+    if (entries.empty())
+    {
+        return;
+    }
+
+    for (auto entry: entries)
+    {
+        std::string key = kfvKey(entry);
+        std::string op = kfvOp(entry);
+
+        std::size_t delimiter_pos = key.find_first_of("|");
+        auto domain_id_str = key.substr(0, delimiter_pos);
+        std::string unique_ip_ifnames;
+
+        memset(&cfg_info, 0, sizeof(mclag_unique_ip_cfg_info));
+
+        count++;
+        SWSS_LOG_NOTICE("mclag unique ip interface Key %s passed", key.c_str());
+
+        unique_ip_ifnames = key.substr(delimiter_pos+1);
+        if (unique_ip_ifnames.empty())
+        {
+            SWSS_LOG_ERROR("Invalid Key %s Format. No unique ip ifname specified", key.c_str());
+            continue;
+        }
+
+        if(op == "SET")
+        {
+            cfg_info.op_type = MCLAG_CFG_OPER_ADD;
+        }
+        else
+        {
+            cfg_info.op_type = MCLAG_CFG_OPER_DEL;
+        }
+
+        memcpy(cfg_info.mclag_unique_ip_ifname, unique_ip_ifnames.c_str(), unique_ip_ifnames.size());
+
+        SWSS_LOG_NOTICE("optype:%d mclag_unique_ip_ifname:%s", cfg_info.op_type, cfg_info.mclag_unique_ip_ifname);
+
+        if (MCLAG_MAX_SEND_MSG_LEN - infor_len < (sizeof(struct mclag_unique_ip_cfg_info)) )
+        {
+            cfg_msg_hdr = reinterpret_cast<mclag_msg_hdr_t *>(static_cast<void *>(infor_start));
+            cfg_msg_hdr->version = 1;
+            cfg_msg_hdr->msg_len = (unsigned short)infor_len;
+            cfg_msg_hdr->msg_type = MCLAG_SYNCD_MSG_TYPE_CFG_MCLAG_UNIQUE_IP;
+
+            SWSS_LOG_NOTICE("mclagsycnd send msg to iccpd, msg_len =%d, msg_type =%d count : %d",
+                    cfg_msg_hdr->msg_len, cfg_msg_hdr->msg_type, count);
+
+            write = ::write(getConnSocket(), infor_start, cfg_msg_hdr->msg_len);
+
+            if (write <= 0)
+            {
+                SWSS_LOG_ERROR("mclagsycnd to ICCPD, mclag unique ip cfg send, buffer full; write to m_connection_socket failed");
```

Comment:
should it just continue after the error?  bail out? or skip current iteration other than just log the error?